qiskit_nature/qmolecule.py

```diff
@@ -73,6 +73,10 @@ def __init__(self, filename=None):
         self.mo_coeff_b = None  # v2
         self.orbital_energies = None
         self.orbital_energies_b = None  # v2
+        self.mo_occ = None  # v3
+        self.mo_occ_b = None  # v3
+
+        self.energy_shift = 0.0  # v3
```

Comment:
See comment in driver around occupied.

I think I would rather keep the QMolecule as some driver raw result and put this inside another class that has the integrals in block (or paired) spin format. My current assumption is that we have 'default' transform somewhere, maybe as method in the class rather than in QMolecule. where they are now, which converts them from the driver form into block spin. I guess it should be possible to go directly to paired spin format too, but if the block spin is more the default/buil-in transform then, at that is what all the transforms work on, then the SeniortiyZero is a transform across the formats and my need to be the last one in sequence (assuming we don't do FreezeCore equivalent etc on the paired spin format, which presumably is doable, but more effort.

The energies can then be accumulated in this extra class. As well as things like num particles tracked as they are removed from the problem too. This way it may be easier to split out as separate properties later too.

Reply:
Since we decided to postpone the refactoring of everything which ranges from drivers to the transforms I think it is okay to add this here in an intermediate step. I need to talk to Panos to clarify some aspects of the SeniorityZeroTransformer because (if I recall correctly) the last discussion we had on this topic made everything seem more complicated than it really needs to be.

Reply:
I have made this a dictionary now. This allows us to keep track of where an energy shift came from while also easily summing over the `dict.values()` to get the total energy shift. We can then refactor this into a nicer handling when looking into the drivers and `QMolecule` for a later release.